dvc/utils/fs.py

```diff
@@ -36,7 +37,11 @@ def get_mtime_and_size(path, fs):
     if fs.isdir(path):
         size = 0
         files_mtimes = {}
-        for file_path in fs.walk_files(path):
+        if fs.scheme == Schemes.LOCAL and fs.repo:
+            walk_iterator = fs.repo.dvcignore(fs.walk(path), walk_files=True)
+        else:
+            walk_iterator = fs.walk_files(path)
+        for file_path in walk_iterator:
```

Comment:
hm, okay, here is where getting `dvcignore` is a bit of a challenge. I think we should pass it explicitly here instead of relying on `fs.repo`, which will go away soon. Please take a look if it is simple to do that right now, and if not - I'll adjust it in my PR on top.

```suggestion
        for fpath in dvcignore.walk_files(fs, path):
```